src/v2i-hub/RSUHealthMonitorPlugin/src/RSUHealthMonitorPlugin.cpp

```diff
@@ -0,0 +1,97 @@
+#include "RSUHealthMonitorPlugin.h"
+
+using namespace RSUHealthMonitor;
+using namespace tmx::utils;
+
+namespace RSUHealthMonitor
+{
+
+    RSUHealthMonitorPlugin::RSUHealthMonitorPlugin(const std::string &name) : PluginClient(name)
+    {
+        _rsuWorker = std::make_shared<RSUHealthMonitorWorker>();
+        _rsuStatusTimer = make_unique<ThreadTimer>();
+        UpdateConfigSettings();
+
+        // Send SNMP call to RSU periodically at configurable interval.
+        _timerThId = _rsuStatusTimer->AddPeriodicTick([this]()
+                                                      {
+            // Periodic SNMP call to get RSU status based on RSU MIB version 4.1
+            auto rsuStatusJson =  _rsuWorker->getRSUStatus(_rsuMibVersion, _rsuIp, _snmpPort, _securityUser, _authPassPhrase, _securityLevel, SEC_TO_MICRO);
+            PLOG(logINFO) << "Updating _interval: " << _interval;
+            //Broadcast RSU status periodically at _interval
+            BroadcastRSUStatus(rsuStatusJson); },
+                                                      std::chrono::milliseconds(_interval * SEC_TO_MILLI));
+        _rsuStatusTimer->Start();
```

Comment:
Can we move this start to when the Plugin is enabled on config changes.

Reply:
Addressed https://github.com/usdot-fhwa-OPS/V2X-Hub/pull/564/commits/a49366ea8030afa8c668ec5879cf8e2d5cf69065